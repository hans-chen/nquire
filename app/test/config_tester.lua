#!/usr/bin/lua

package.path = "../?.lua;./?.lua"
 
require "strict"
require "getopt"
require "format"
require "log"
require "config"
require "base64"
require "evq"
require "config_node"
require "sys"
require "misc"

require "tester"
th = Tester.new()

os.execute( "rm config_tester.conf" )

logf_init(LG_INF, true, true)
--logf_init(LG_DMP, true, true)
evq = Evq.new()
config = Config.new("config_tester.schema", "config_tester.conf" )

function tst( s, ... )
	th:start("config:set_config_item('" .. s .. "')")

	print("Before:")
	for i,item in ipairs({...}) do
		print(item .. " = \"" .. config:get(item) .. "\"")
	end
	config:set_config_item(s)
	print("After:")
	for i,item in ipairs({...}) do
		print(item .. " = \"" .. config:get(item) .. "\"")
	end
end

tst( "/group1/number1 = 3", "/group1/number1")
tst( "/group1/enum1 = \"three\"", "/group1/enum1" )
tst( "/group1/enum1 = \"one\" ; /group1/string1 = \"blabla\"", "/group1/enum1", "/group1/string1" )
tst( "/group1/enum1 = \"two\" /group1/string1 = \"blat\"", "/group1/enum1", "/group1/string1") 
tst( "/group1/string1 = \"a\\bc\"", "/group1/string1")
tst( "/group1/string1 = \"\\xabq\"", "/group1/string1")
tst( "/group1/string1 = \"\\xaq\"", "/group1/string1")
tst( '/group1/string1 = "a\\"b"', "/group1/string1")
tst( "/group1/string1 = \"\\n\"", "/group1/string1")

th:start("save_db")

config:save_db( "config_tester.conf1" )

-- and print the new configfile contents for verification
print("Configfile:")
local fd = io.open("config_tester.conf1", "r")
local txt = fd:read("*all")
fd:close()
print(txt)

local function on_group1_string1(node, data)
	print("on_group1_string1, data=" .. data .. ", value=" .. node:get())
end

config:add_watch("/group1/string1", "set", on_group1_string1,"mydata")

print("before set: bla")
config:set( "/group1/string1", "blabla" )
print("after set, now pop:")
for i=1,14 do
evq:pop(true)
end

print("before set: direct")
config:set( "/group1/string1", "direct", true )
print("after set")
evq:pop()
evq:pop()
