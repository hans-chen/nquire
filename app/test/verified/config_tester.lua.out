# *** Running module/config_tester.lua

[wrn] : No config file found. Using defaults.

# *** Start test 'config:set_config_item('/group1/number1 = 3')'
Before:
/group1/number1 = "5"
After:
/group1/number1 = "3"

# *** Start test 'config:set_config_item('/group1/enum1 = "three"')'
Before:
/group1/enum1 = "two"
After:
/group1/enum1 = "three"

# *** Start test 'config:set_config_item('/group1/enum1 = "one" ; /group1/string1 = "blabla"')'
Before:
/group1/enum1 = "three"
/group1/string1 = "abcd"
After:
/group1/enum1 = "one"
/group1/string1 = "blabla"

# *** Start test 'config:set_config_item('/group1/enum1 = "two" /group1/string1 = "blat"')'
Before:
/group1/enum1 = "one"
/group1/string1 = "blabla"
After:
/group1/enum1 = "two"
/group1/string1 = "blat"

# *** Start test 'config:set_config_item('/group1/string1 = "a\bc"')'
Before:
/group1/string1 = "blat"
After:
/group1/string1 = "a\bc"

# *** Start test 'config:set_config_item('/group1/string1 = "\xabq"')'
Before:
/group1/string1 = "a\bc"
After:
/group1/string1 = "«q"

# *** Start test 'config:set_config_item('/group1/string1 = "\xaq"')'
Before:
/group1/string1 = "«q"
After:
/group1/string1 = "\xaq"

# *** Start test 'config:set_config_item('/group1/string1 = "a\"b"')'
Before:
/group1/string1 = "\xaq"
After:
/group1/string1 = "a"b"

# *** Start test 'config:set_config_item('/group1/string1 = "\n"')'
Before:
/group1/string1 = "a"b"
After:
/group1/string1 = "
"

# *** Start test 'save_db'
Configfile:

# Settings


# group one

/group1/enum1 = "two"
/group1/number1 = 3
/group1/string1 = "\n"

# End


before set: bla
after set, now pop:
on_group1_string1, data=mydata, value=blabla
before set: direct
on_group1_string1, data=mydata, value=direct
after set
[wrn] : No config file found. Using defaults.
[inf] : Loading config file cit.conf
/dev/mifare/key_A = FFFFFFFFFFFF
/dev/mifare/relevant_sectors = 0:0,15:2
range("1") = true
range("0") = true
range("15") = true
range("16") = false
range("0,15") = true
range("0,16") = false
range("0:0") = true
range("15:2") = true
range("0:3") = false
range("15:3") = false
range("16:0") = false
range("0:0,0:1,0:2") = true
range("15:0,15:1,15:2") = true
